tsdb/head.go

```diff
@@ -307,6 +309,35 @@ func newHeadMetrics(h *Head, r prometheus.Registerer) *headMetrics {
 	return m
 }
 
+// HeadStats are the statistics for the head component of the DB.
+type HeadStats struct {
+	WALReplayStatus *WALReplayStatus
+}
+
+// NewHeadStats returns a new HeadStats object.
+func NewHeadStats() *HeadStats {
+	return &HeadStats{
+		WALReplayStatus: &WALReplayStatus{},
+	}
+}
+
+// WALReplayStatus contains status information about the WAL replay.
+type WALReplayStatus struct {
+	sync.RWMutex
+	Min     int
+	Max     int
+	Current int
+}
+
+// GetWALReplayStatus returns the WAL replay status information.
+func (s *WALReplayStatus) GetWALReplayStatus() WALReplayStatus {
+	s.RLock()
+	x := s
```

Comment:
```suggestion
	x := *s
```

Otherwise we still might have race when dereferencing on line 338 🙈 

But it's much better than it was!

Reply:
Actually linter already mention this: 
![image](https://user-images.githubusercontent.com/6950331/120528521-772ebe80-c3d3-11eb-9fe9-0ba9e2665948.png)

Robots take my job away 😢

Reply:
It (still) complains about copying the lock value, is that a problem?

Reply:
Yeah, the linter is complaining about it now. Those darn robots...